frontend/packages/dev-console/src/components/pipelines/PipelineVisualizationTask.tsx

```diff
@@ -96,7 +102,11 @@ const TaskComponent: React.FC<TaskProps> = (props) => {
       >
         <div className="odc-pipeline-vis-task__content">
           <div className={cx('odc-pipeline-vis-task__title', { 'is-text-center': !status })}>
-            {task.metadata ? task.metadata.name : ''}
+            {props.name
```

Comment:
same as above

Reply:
The variable `task` is obtained from `props.task.data`. If props is destructured, it will create a variable name conflict with `task` in use in other places. cc @karthikjeeyar

Reply:
How about `props.name || _.get(task, ['metadata', 'name'], '')`?